Sources/CoreData/Concurrency/AsyncObjectListRepository.swift

```diff
@@ -0,0 +1,25 @@
+//
+// AppwiseCore
+// Copyright © 2023 Wisemen
+//
+
+import Foundation
+
+@available(iOS 13.0, *)
+public protocol AsyncObjectListRepository: ObjectListRepository {
+	func refresh() async throws -> [ObjectType]
+}
+
+@available(iOS 13.0, *)
+public extension AsyncObjectListRepository {
+	func refresh() async throws -> [ObjectType] {
+		throw Cancelled()
```

Comment:
That would require iOS 13+. Isn't a big issue, but keep it in mind.

Reply:
Does that error have any side effects? Because note that we're not starting a task, or cancelling anything. TBH we're not doing anything, which isn't exactly the same as cancelled 🤷

Reply:
- Deprecated own cancelled error in favour of swift's error
- Changed `refresh` default implementation to assert (easily overlooked issue)
- Changed `refresh` default implementation to throw a private error, so the implementation keeps working as expected (it still calls the handler)

Reason for the 1st item, is that I noticed that the cancelled error is being abused/reused for other purposes in some codebases, which it was never meant for.